core/src/transport.rs

```diff
@@ -57,6 +57,17 @@ pub use self::optional::OptionalTransport;
 pub use self::upgrade::Upgrade;
 
 static NEXT_LISTENER_ID: AtomicUsize = AtomicUsize::new(1);
+#[derive(Debug, Default, Copy, Clone)]
+pub enum PortUse {
+    #[default]
+    New,
+    Reuse,
+}
```

Comment:
Why is `New` the default? In a peer-to-peer network, I would expect `Reuse` to be the default, given that it allows:

- discovering ones public address
- keeping NAT mappings alive across connections
- hole punching